src/Module/Settings/Uexport.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * @file src/Modules/Settings/Uexport.php
+ */
+
+namespace Friendica\Module\Settings;
+
+use Friendica\App;
+use Friendica\App\Arguments;
+use Friendica\BaseModule;
+use Friendica\Core\Hook;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Core\System;
+use Friendica\Database\DBA;
+use Friendica\Database\DBStructure;
+use Friendica\Module\BaseSettingsModule;
+
+/**
+ * Module to export user data
+ **/
+class Uexport extends BaseSettingsModule
+{
+	/**
+	 * Handle the request to export data.
+	 * At the moment one can export two different data set
+	 * 1. The profile data that can be used by uimport to resettle
+	 *    to a different Friendica instance
+	 * 2. The entire data-set, profile plus postings
+	 *
+	 * If there is an action required through the URL / path, react
+	 * accordingly and export the requested data.
+	 **/
+	public static function content()
+	{
+		parent::content();
+		$args = self::getClass(Arguments::class);
+		if ($args->getArgc() == 3) {
+			// @TODO Replace with router-provided arguments
+			$action = $args->get(2);
+			header("Content-type: application/json");
+			header('Content-Disposition: attachment; filename="' . $a->user['nickname'] . '.' . $action . '"');
+			switch ($action) {
+				case "backup":
+					SELF::uexport_all($a);
+					exit();
+					break;
+				case "account":
+					SELF::uexport_account($a);
+					exit();
+					break;
+				default:
+					exit();
+			}
+		}
```

Comment:
Please move this block to a separate `public static function rawContent()` method.

Additionally, `self` isn't supposed to be capitalized and please normalize the quoting style one way or another.

Reply:
Why should that class function be public?

Reply:
Because it's eventually called from `App\Module` which doesn't inherit from this module class. As such, we can't use "protected" either. With `content()` and `post()` it's one of the 3 base public methods modules have to implement at least one of them to be usable.